addons/web/static/src/core/utils/template_inheritance.js

```diff
@@ -0,0 +1,196 @@
+/** @odoo-module **/
+
+import { _t } from "@web/core/l10n/translation";
```

Comment:
@xmo-odoo gave me an idea.
Currently, the xpath instructions are self-referential: they apply on the new document that is being produced by the xpath instructions.

```xml
<!-- source xml -->
<div>
  <h1/>
</div>

<!-- inheritance xml -->
<xpath expr="/div/h1" position="before">
    <h1 />
</xpath>

<xpath expr="/div/h1" position="after">
    <div />
</xpath>
 ```
In this example, the div will be added after the added h1, which is fine for "manual" inheritance. When computing inheritance, an hypothetic (but not so hypothetic) algo should keep track of the positions and tags of known elements, which is very cumbersome.

The idea is to put a capture_first attribute on the data nodes, that will make the inheritance algo capture the target node in the source arch first, and only then apply the instructions.

This is probably out of scope for this task, but not that far away